Utilities/Dox/PythonScripts/FileManGlobalDataParser.py

```diff
@@ -790,59 +790,71 @@ def run(args):
   from InitCrossReferenceGenerator import parseCrossRefGeneratorWithArgs
   from FileManDataToHtml import FileManDataToHtml
 
+  # Ensure that output directory exists
+  if not os.path.exists(os.path.join(args.outDir, "dox")):
+    os.makedirs(os.path.join(args.outDir, "dox"))
+
   crossRef = parseCrossRefGeneratorWithArgs(args)
-  _doxURL = getDOXURL(args.local)
-  _vivianURL = getViViaNURL(args.local)
-  glbDataParser = FileManGlobalDataParser(args.MRepositDir, crossRef)
-  assert '0' in glbDataParser.allFiles and '1' in glbDataParser.allFiles and set(args.fileNos).issubset(glbDataParser.allFiles)
 
-  # Populate glbDataParser.globalLocationMap
+  # Populate glbDataParse
+  glbDataParser = FileManGlobalDataParser(args.MRepositDir, crossRef)
   glbDataParser.parseZWRGlobalFileBySchemaV2(glbDataParser.allFiles['1']['path'][0], '1', '^DIC(')
-  for fileNo in args.fileNos:
-    assert fileNo in glbDataParser.globalLocationMap
   del glbDataParser.outFileManData['1']
   glbDataParser.outdir = args.outDir
+  glbDataParser.patchDir = args.patchRepositDir
 
-  if not os.path.exists(os.path.join(args.outDir, "dox")):
-    os.makedirs(os.path.join(args.outDir, "dox"))
+  _doxURL = getDOXURL(args.local)
+  _vivianURL = getViViaNURL(args.local)
+  htmlGen = FileManDataToHtml(crossRef, glbDataParser.schemaParser,
+                              args.outDir, _doxURL, _vivianURL)
 
-  glbDataParser.patchDir = args.patchRepositDir
-  htmlGen = FileManDataToHtml(crossRef, glbDataParser.schemaParser, args.outDir, _doxURL, _vivianURL)
-  isolatedFiles = glbDataParser.schemaParser.isolatedFiles
-  if not args.all or set(args.fileNos).issubset(isolatedFiles):
+  if not args.all:
+    isolatedFiles = glbDataParser.schemaParser.isolatedFiles
+    assert set(args.fileNos).issubset(isolatedFiles)
```

Comment:
Trying to run over only file 101 returned a failure at this assertion.  I wonder if this assertion is necessary anymore.

> python FileManGlobalDataParser.py -mr /opt/VistA-M/ -pr /opt/VistA -o /opt/viv-out/ -lf /tmp/ -f 101
<....>
> Traceback (most recent call last):
>   File "FileManGlobalDataParser.py", line 905, in <module>
>     main()
>   File "FileManGlobalDataParser.py", line 902, in main
>     run(result)
>   File "FileManGlobalDataParser.py", line 813, in run
>     assert set(args.fileNos).issubset(isolatedFiles)
> AssertionError
>

Reply:
Fixed